test/DateYearDurationTest.cpp

```diff
@@ -0,0 +1,977 @@
+                                                     int minutes,
+                                                     double seconds) {
+  // to milliseconds
+  auto millisecDays = static_cast<long long>(days) * 24 * 60 * 60 * 1000;
+  auto millisecHours = static_cast<long long>(hours) * 60 * 60 * 1000;
+  auto millisecMinutes = static_cast<long long>(minutes) * 60 * 1000;
+  auto millisecSeconds = static_cast<long long>(std::round(seconds * 1000));
+  auto totalMilliseconds =
+      millisecDays + millisecHours + millisecMinutes + millisecSeconds;
+
+  // from milliseconds
+  long long remainingMilliseconds = totalMilliseconds;
+  days = remainingMilliseconds / (24 * 60 * 60 * 1000);
+  remainingMilliseconds %= 24 * 60 * 60 * 1000;
+  hours = remainingMilliseconds / (60 * 60 * 1000);
+  remainingMilliseconds %= 60 * 60 * 1000;
+  minutes = remainingMilliseconds / (60 * 1000);
+  remainingMilliseconds %= 60 * 1000;
+  seconds = remainingMilliseconds / 1000.0;
+  return {days, hours, minutes, seconds};
+}
+
+//______________________________________________________________________________
+TEST(DayTimeDuration, checkInternalConversionForLargeValues) {
+  int maxDays = static_cast<int>(DayTimeDuration::maxDays);
+  ad_utility::SlowRandomIntGenerator randomDay{1000000, maxDays};
+  ad_utility::SlowRandomIntGenerator randomHour{22, 1000000};
+  ad_utility::SlowRandomIntGenerator randomMinute{55, 1000000};
+  ad_utility::RandomDoubleGenerator randomSecond{60.000, 99999.9990};
+
+  size_t i = 0, max = 1024;
+  while (i < max) {
+    int randDay = randomDay();
+    int randHour = randomHour();
+    int randMinute = randomMinute();
+    int randSeconds = randomSecond();
+
+    DayTimeDuration::DurationValue dv1 =
+        toAndFromMilliseconds(randDay, randHour, randMinute, randSeconds);
+
+    if (dv1.days_ > maxDays) {
+      ASSERT_THROW(DayTimeDuration(DayTimeDuration::Type::Positive, randDay,
+                                   randHour, randMinute, randSeconds),
+                   DurationOverflowException);
+    } else {
+      const auto& dv2 =
+          DayTimeDuration(DayTimeDuration::Type::Positive, randDay, randHour,
+                          randMinute, randSeconds)
+              .getValues();
+      ASSERT_EQ(dv1.days_, dv2.days_);
+      ASSERT_EQ(dv1.hours_, dv2.hours_);
+      ASSERT_EQ(dv1.minutes_, dv2.minutes_);
+      EXPECT_NEAR(dv1.seconds_, dv2.seconds_, 0.001);
+    }
+    i++;
+  }
+}
+
+//______________________________________________________________________________
+TEST(DayTimeDuration, checkParseAndGetString) {
+  // Set the lower limit to 1, 0's will be mostly (some special cases are
+  // tested directly above) ignored when constructing a xsd:dayTimeDuration
+  // string.
+  ad_utility::SlowRandomIntGenerator randomDay{1, 1048574};
+  ad_utility::SlowRandomIntGenerator randomHour{1, 23};
+  ad_utility::SlowRandomIntGenerator randomMinute{1, 59};
```

Comment:
As this test duplicates the code for parsing somewhat, it is a little bit redundant.
Maybee you should have some hardcoded tests also (except if you already have tthem somewhere else, in that case the test case name here is off).

Reply:
The are, i replaced the with hard coded ones